go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -297,28 +298,33 @@ func (n *NodeController) hybridOverlayNodeUpdate(node *kapi.Node) error {
 	// Distributed Router MAC ARP responder flow; responds to ARP requests by OVN for
 	// any IP address within this node's assigned subnet and returns our hybrid overlay
 	// port's MAC address.
-	flows = append(flows,
-		fmt.Sprintf("cookie=0x%s,table=0,priority=100,arp,in_port=ext,arp_tpa=%s,"+
-			"actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],"+
-			"mod_dl_src:%s,"+
-			"load:0x2->NXM_OF_ARP_OP[],"+
-			"move:NXM_NX_ARP_SHA[]->NXM_NX_ARP_THA[],"+
-			"load:0x%s->NXM_NX_ARP_SHA[],"+
-			"move:NXM_OF_ARP_TPA[]->NXM_NX_REG0[],"+
-			"move:NXM_OF_ARP_SPA[]->NXM_OF_ARP_TPA[],"+
-			"move:NXM_NX_REG0[]->NXM_OF_ARP_SPA[],"+
-			"IN_PORT",
-			cookie, cidr.String(), drMAC.String(), drMACRaw))
-	// Send all flows for the remote node's assigned subnet to that node via the VXLAN tunnel.
-	// Windows hybrid overlay implementation requires that we set the destination MAC address
-	// to the node's Distributed Router MAC.
-	flows = append(flows,
-		fmt.Sprintf("cookie=0x%s,table=0,priority=100,ip,nw_dst=%s,"+
-			"actions=load:%d->NXM_NX_TUN_ID[0..31],"+
-			"set_field:%s->tun_dst,"+
-			"set_field:%s->eth_dst,"+
-			"output:"+extVXLANName,
-			cookie, cidr.String(), hotypes.HybridOverlayVNI, nodeIP.String(), drMAC.String()))
+	for index, cidr := range cidrs {
+		//TODO : DUAL-STACK : Neighbour discovery for ipv6?
+		if !utilnet.IsIPv6CIDR(cidr) {
+			flows = append(flows,
+				fmt.Sprintf("cookie=0x%s,table=0,priority=100,arp,in_port=ext,arp_tpa=%s,"+
+					"actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],"+
+					"mod_dl_src:%s,"+
+					"load:0x2->NXM_OF_ARP_OP[],"+
+					"move:NXM_NX_ARP_SHA[]->NXM_NX_ARP_THA[],"+
+					"load:0x%s->NXM_NX_ARP_SHA[],"+
+					"move:NXM_OF_ARP_TPA[]->NXM_NX_REG0[],"+
+					"move:NXM_OF_ARP_SPA[]->NXM_OF_ARP_TPA[],"+
+					"move:NXM_NX_REG0[]->NXM_OF_ARP_SPA[],"+
+					"IN_PORT",
+					cookie, cidrs[index].String(), drMAC.String(), drMACRaw))
+		}
+		// Send all flows for the remote node's assigned subnet to that node via the VXLAN tunnel.
+		// Windows hybrid overlay implementation requires that we set the destination MAC address
+		// to the node's Distributed Router MAC.
+		flows = append(flows,
+			fmt.Sprintf("cookie=0x%s,table=0,priority=100,ip,nw_dst=%s,"+
+				"actions=load:%d->NXM_NX_TUN_ID[0..31],"+
+				"set_field:%s->tun_dst,"+
+				"set_field:%s->eth_dst,"+
+				"output:"+extVXLANName,
+				cookie, cidrs[index].String(), hotypes.HybridOverlayVNI, nodeIPs[index].String(), drMAC.String()))
```

Comment:
don't assume `cidrs` and `nodeIPs` are in the same order. You can use `util.MatchIPFamily` to find the cidr that goes with an IP.

Reply:
Done